.gcloudignore

```diff
@@ -15,6 +15,3 @@
 
 # Node.js dependencies:
 node_modules/
-
-# API Secrets
-*.json
```

Comment:
Why this?

Reply:
There are no keyfiles in the repo, but package.json has to be uploaded for Cloud Functions to download dependencies. I suppose this could just exclude package.json if the key json files are still used for local development? Not sure if that still happens

Reply:
We can specifically ban https://github.com/ampproject/error-tracker/blob/master/keys.js#L8, then? This was added so my deploy environment wouldn't be different from anyone else's. It forced me to make the `keys.js` startup script so that anyone could deploy.

Reply:
I guess we can, if it's not needed? It's not clear to me what these files are used for. Are paths to these files provided as environment variables when you deploy? In `ampproject/amp-github-apps` I configured all the apps in the repo to deploy when a tag is pushed using Cloud Build so that environments and auth and such wasn't dependent on local environment conditions. I think that might make sense here.

Reply:
Cloud Functions can have their own environment variables set. I believe we can just take the contents of these keyfiles and bind them to the cloud function deployments

Reply:
There are authorization keys necessary to post to the other project buckets (ads/user errors). They're rather large (2kb), so I have them stored as actual files instead of ENV vars.

Reply:
Cloud Functions have limited filesystem access
1) They can access the read-only file directory that is uploaded with the function. This would mean deploying the keyfiles with the function itself. Not too bad
2) They can access a writable `/tmp` directory (which is in-memory), and so the app could fetch and write keyfiles on startup, and read from that filesystem
3) They can read directly from Cloud Storage on startup
4) They can read the JSON file contents from an environment variable

For the sake of simplicity I'd lean toward 1. Do you see any downsides to this? I'm not familiar with how the GAE app was deployed, so maybe this is the current behavior

Reply:
The keys are actually stored in Cloud Storage, so it seems like we could just move the import logic from the pre script into the actual app.

Reply:
I'd like to clean up `keys.js` into a more structured form, but I think it would be better to do that in a separate PR/thread (including a discussion about keeping it at all versus just providing a way for the deployed app to get the keys). At present, deploying the app with the `.json` key files included is sufficient for this to function. Does that SGTY?

Reply:
I'd prefer to provide a very simple `KeyStorage` class that uses logic like: https://github.com/ampproject/amp-github-apps/blob/master/owners/src/api/cloud_storage.js#L64